tools/app.hpp

```diff
@@ -128,34 +129,45 @@ namespace arg {
         {
             auto* wand = app->add_option("-w,--wand", m_wand_data_path, "WAND data filename");
             auto* scorer =
-                app->add_option("-s,--scorer", m_scorer, "Query processing algorithm")->needs(wand);
+                app->add_option("-s,--scorer", m_params.m_name, "Query processing algorithm")->needs(wand);
```

Comment:
This doesn't work as it currently stands, as m_params is of type `std::optional<ScorerParams>` and so `m_name` cannot be accessed. Same for the other arguments.